packages/builders/src/messages/embed/Assertions.ts

```diff
@@ -1,99 +1,55 @@
-	.setValidationEnabled(isValidationEnabled);
-
-export const fieldValuePredicate = s
-	.string()
-	.lengthGreaterThanOrEqual(1)
-	.lengthLessThanOrEqual(1_024)
-	.setValidationEnabled(isValidationEnabled);
-
-export const fieldInlinePredicate = s.boolean().optional();
-
-export const embedFieldPredicate = s
-	.object({
-		name: fieldNamePredicate,
-		value: fieldValuePredicate,
-		inline: fieldInlinePredicate,
-	})
-	.setValidationEnabled(isValidationEnabled);
-
-export const embedFieldsArrayPredicate = embedFieldPredicate.array().setValidationEnabled(isValidationEnabled);
-
-export const fieldLengthPredicate = s.number().lessThanOrEqual(25).setValidationEnabled(isValidationEnabled);
-
-export function validateFieldLength(amountAdding: number, fields?: APIEmbedField[]): void {
-	fieldLengthPredicate.parse((fields?.length ?? 0) + amountAdding);
-}
-
-export const authorNamePredicate = fieldNamePredicate.nullable().setValidationEnabled(isValidationEnabled);
-
-export const imageURLPredicate = s
-	.string()
-	.url({
-		allowedProtocols: ['http:', 'https:', 'attachment:'],
-	})
-	.nullish()
-	.setValidationEnabled(isValidationEnabled);
-
-export const urlPredicate = s
-	.string()
-	.url({
-		allowedProtocols: ['http:', 'https:'],
-	})
-	.nullish()
-	.setValidationEnabled(isValidationEnabled);
-
-export const embedAuthorPredicate = s
-	.object({
-		name: authorNamePredicate,
-		iconURL: imageURLPredicate,
-		url: urlPredicate,
-	})
-	.setValidationEnabled(isValidationEnabled);
-
-export const RGBPredicate = s
-	.number()
-	.int()
-	.greaterThanOrEqual(0)
-	.lessThanOrEqual(255)
-	.setValidationEnabled(isValidationEnabled);
-export const colorPredicate = s
-	.number()
-	.int()
-	.greaterThanOrEqual(0)
-	.lessThanOrEqual(0xffffff)
-	.or(s.tuple([RGBPredicate, RGBPredicate, RGBPredicate]))
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
+export const namePredicate = z.string().min(1).max(256);
 
-export const descriptionPredicate = s
+export const iconURLPredicate = z
 	.string()
-	.lengthGreaterThanOrEqual(1)
-	.lengthLessThanOrEqual(4_096)
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
+	.url()
+	.refine(refineURLPredicate(['http:', 'https:', 'attachment:']), {
+		message: 'Invalid protocol for icon URL. Must be http:, https:, or attachment:',
+	});
 
-export const footerTextPredicate = s
+export const URLPredicate = z
 	.string()
-	.lengthGreaterThanOrEqual(1)
-	.lengthLessThanOrEqual(2_048)
-	.nullable()
-	.setValidationEnabled(isValidationEnabled);
-
-export const embedFooterPredicate = s
+	.url()
+	.refine(refineURLPredicate(['http:', 'https:']), { message: 'Invalid protocol for URL. Must be http: or https:' });
+
+export const embedFieldPredicate = z.object({
+	name: namePredicate,
+	value: z.string().min(1).max(1_024),
+	inline: z.boolean().optional(),
+});
+
+export const embedAuthorPredicate = z.object({
+	name: namePredicate,
+	icon_url: iconURLPredicate.optional(),
+	proxy_icon_url: iconURLPredicate.optional(),
```

Comment:
funnily enough, I don't think bots can ever set this field, and its just a discord-set field 👀

Reply:
I'll test this later

Reply:
I believe that falls [here](https://discord.com/developers/docs/resources/message#create-message-limitations):
> For the embed object, you can set every field except `type` [...], `provider`, `video`, and any `height`, `width`, or `proxy_url` values for images.